django/db/models/fields/__init__.py

```diff
@@ -218,7 +219,12 @@ def __init__(
         self.unique_for_year = unique_for_year
         if isinstance(choices, collections.abc.Iterator):
             choices = list(choices)
-        self.choices = choices
+        if isinstance(choices, ChoicesMeta):
+            self._choices = choices  # Store origin for reference
```

Comment:
Perhaps this attribute can have a more meaningful name?
`self._choices_class`, `self._choices_meta`, something in this vein - to avoid future confusions with `self.choices`.

Reply:
Sure, I would opt for `self._choices_class` as it's not really a meta class but an instance of meta. Originally chose the name to stay in line with above `_verbose_name`